### DISCUSSION TITLE
Using existing SMTP server?

### DISCUSSION THREAD

**Discussion by LouisIV** [postalserver/postal#1587] [Help with using Postal]
I can’t seem to find any documentation for using postal with an existing SMTP server. In this case ImprovMX. Postal looks great and I would love to plug it in instead of the shim we have now.

**Comment by willpower232** [comment]
I am not sure what benefit Postal would be to ImprovMX, can you explain a bit more about what you are trying to do?

**Reply by LouisIV** [reply]
I'm hoping to take advantage of the built in tracking and tools to help protect my sender reputation

**Reply by willpower232** [reply]
What does the shim you have now do?

**Reply by LouisIV** [reply]
I guess shim isn't the right word. It's just turning some of our API requests into emails. It's not ideal for us and doesn't check things like deliverability, which to my understanding postal does

**Reply by LouisIV** [reply]
Is there any way to swap out the internal SMTP server for something else? Maybe with some fancy docker compose networking?

**Reply by willpower232** [reply]
Postals entire job is to send emails, I don't think it tracks deliverability specifically but if you already have your own API requests, you could replace those with API requests to Postal, store the returned message ID and then listen to webhooks so you can see what happened to that message within your application.

The internal SMTP server receives emails only, the sending part is done via a worker process. I don't think you could (or should) make it a modular component because I don't think there would be a lot of benefit.

Does that answer how you could use Postal?
